rat/variables.py

```diff
@@ -151,10 +153,24 @@ class Param:
     subscript: Subscript = None
     lower: float = float("-inf")
     upper: float = float("inf")
+    constraints_set: bool = False
 
     def set_constraints(self, lower, upper):
```

Comment:
how does `set_constraints` work with `Data`?

Reply:
I wasn't thinking about constraints on data. We could certainly add those. For now I'm assuming no mistakes :P.

Reply:
Oh sorry github diff viewer made it look like `set_constraints` existed for `Data`. I agree with you